File: python/number_palindrome/number_palindrome.py
import random
import math

max_n = 10**5
n = random.randrange(max_n)

max_log = int(math.floor(math.log(n, 10)))
odd_log = max_log % 2 == 0
half_max_log = int(math.floor(max_log / 2))

top_low_range = half_max_log if odd_log else half_max_log + 1
range_low = xrange(0, top_low_range)
range_high = xrange(max_log, half_max_log, -1)

def get_indices(n, index_range):
  return [(n % (10**(i + 1)) / (10**i)) for i in index_range]

digits_low = get_indices(n, range_low)
digits_high = get_indices(n, range_high)

print n
print digits_low, digits_high
print digits_low == digits_high

File: javascript/number_palindrome/number_palindrome.js
const max_n = Math.pow(10, 5);
const n = Math.floor(Math.random() * max_n);

const max_log = Math.floor(Math.log10(n));
const odd_log = max_log % 2 == 0;
const half_max_log = Math.floor(max_log / 2);

const range_low_min = 0;
const range_low_max = odd_log ? half_max_log - 1 : half_max_log;
const range_high_min = half_max_log + 1;
const range_high_max = max_log;

function get_indices_by_range (n, low, high) {
  const result = [];
  for (let i = low; i <= high; i++) {
    let imod = Math.pow(10, i + 1);
    let idiv = Math.pow(10, i);
    result.push(
      Math.floor((n % imod) / idiv)
    );
  }
  return result;
}

const digits_low = get_indices_by_range(n, range_low_min, range_low_max);
const digits_high = get_indices_by_range(n, range_high_min, range_high_max);
digits_high.reverse();

console.log(n);
console.log(digits_low, digits_high);
console.log(digits_low.toString() == digits_high.toString());
